pantalaimon/daemon.py

```diff
@@ -824,6 +829,64 @@ async def messages(self, request):
             body=await response.read(),
         )
 
+    def _get_upload_and_media_info(self, content_key, content):
+        content_uri = content[content_key]
+
+        try:
+            upload_info = self.upload_info[content_uri]
+        except KeyError:
+            upload_info = self.store.load_upload(self.name, content_uri)
+            if not upload_info:
+                return None, None
+
+        self.upload_info[content_uri] = upload_info
+
+        content_uri = content[content_key]
+        mxc = urlparse(content_uri)
+        mxc_server = mxc.netloc.strip("/")
+        mxc_path = mxc.path.strip("/")
+
+        media_info = self.store.load_media(self.name, mxc_server, mxc_path)
+        if not media_info:
+            return None, None
+
+        self.media_info[(mxc_server, mxc_path)] = media_info
+
+        return upload_info, media_info
+
+    async def _map_decrypted_uri(self, content_key, content, request, client):
+        try:
+            upload_info, media_info = self._get_upload_and_media_info(content_key, content)
+            if not upload_info or not media_info:
+                return await self.forward_to_web(request, token=client.access_token)
```

Comment:
This forwards the request to the server, but every call of the `_map_decrypted_uri()` method will call as well `foward_to_web()` right after this method returns. The call sites for this method seem to expect the happy path exclusively as well, that is they expect the edited content at the bottom of this method.

I think clearing up the goals for this method with some documentation would reveal that the method isn't correctly used or the method isn't doing what it's intending to do in the error cases.

Reply:
I've removed the hidden `forward_to_web()` in `_map_decrypted_uri()` and handled exceptions in the callers.
is this ok?